internal/cpp_doc_generator/cxx_event.rst

```diff
@@ -0,0 +1,97 @@
+Event
+=====
+
+This minimal example showcases how push-notifications can be realized by using services
+with event messaging pattern between two processes. The listener hereby waits for a
+notification from the notifier.
+
+This example is a simplified version of the
+`event example <https://github.com/eclipse-iceoryx/iceoryx2/tree/main/examples/cxx/event>`_.
+You can execute it by opening two terminals and calling:
+
+**Terminal 1:**
+
+.. code-block:: sh
+
+   ./target/ffi/build/examples/cxx/event/example_cxx_event_listener
+
+**Terminal 2:**
+
+.. code-block:: sh
+
+   ./target/ffi/build/examples/cxx/event/example_cxx_event_notifier
+
+
+Listener
+--------
+
+.. code-block:: C++
```

Comment:
I think I need to spent some time on geoffrey to make it work with Rust and restructured text